src/main/java/pariveda/osdu/OsduSimpleClient.java

```diff
@@ -9,30 +9,44 @@
 	//Bring your own token client, gets called in the app.java
 	String token;
 	
-	OsduSimpleClient(String my_url, String my_datapartition, String my_contentType, String my_token){
+	public OsduSimpleClient(String my_url, String my_datapartition, String my_contentType, String my_token){
 		super(my_url, my_datapartition, my_contentType);
 		this.token = my_token;
 	}
 
 	public JSONObject Search(String query) throws IOException{
 		System.out.println("OSDU Search.. Query: "+ query);
-		RequestBody body = RequestBody.create(this.mediaType, query);
-	    Request request = new Request.Builder()
-	      .url(this.url+"/api/search/v2/query")
-	      .method("POST", body)
-	      .addHeader("Content-Type", this.contentType)
-	      .addHeader("data-partition-id", this.dataPartition)
-	      .addHeader("Authorization", this.token)
-	      .build();
-	    
-	    Response response = this.client.newCall(request).execute();
-	    JSONObject resp = new JSONObject(response.body().string());
-	    //System.out.println("Resp in OSDU Simple Client: " +resp);
-	    return resp;
+		if(isValidSearchQuery(query)) {
+			RequestBody body = RequestBody.create(query, this.mediaType);
+			Request request = new Request.Builder()
+			.url(this.url+"/api/search/v2/query")
+			.method("POST", body)
+			.addHeader("Content-Type", this.contentType)
```

Comment:
Do we need both this and the media type? I would think the Request builder automatically appends a content type header based on the associated media type. This could cause them to get out of sync.

Reply:
I did some local testing of the obj, and without it, the request only builds it into the RequestBody. I've elected to leave this alone since this was part of the original commit